solenspipe/bias.py

```diff
@@ -86,15 +152,235 @@ def rdn0(icov,alpha,beta,qfunc,get_kmap,comm,power,nsims,
                 rdn0 += power(qa(X,Ys),qb(A,Bs)) + power(qa(Xs,Y),qb(A,Bs)) \
                         + power(qa(Xs,Y),qb(As,B)) + power(qa(X,Ys),qb(As,B))
             if not(gaussian_sims):
-                Ysp=get_kmap((icov,1,i))
-                Asp=get_kmap((icov,1,i))
-                Bsp = get_kmap((icov,1,i))
+                Ysp=get_kmap((icov,1,j))
+                Asp=get_kmap((icov,1,j))
+                Bsp = get_kmap((icov,1,j))
                 rdn0 += (- power(qa(Xs,Ysp),qb(As,Bsp)) - power(qa(Xs,Ysp),qb(Asp,Bs)))
             else:
                 rdn0 +=  (-power(qa(Xs,Ys),qb(As,Bs)))
     totrdn0 = utils.allreduce(rdn0,comm) 
+
     return totrdn0/nsims
 
+
+    
+
+def rdn0_with_error(icov,alpha,beta,qfunc,get_kmap,comm,power,nsims,
```

Comment:
Do we need three different functions here? Seems at least this one and `rdn0` could be merged.

Reply:
Agreed on this, I think I brought it up earlier with @qujia7